2-Draft-Accepted/RFCxxxx-Secrets-Management.md

```diff
@@ -0,0 +1,150 @@
+---
+RFC: RFCxxxx
+Author: Steve Lee
+Status: Draft
+SupercededBy: N/A
+Version: 0.1
+Area: Security
+Comments Due: July 31st, 2019
+Plan to implement: Yes, PS7
+---
+
+# Simplify and secure use of secrets in scripts
+
+Advanced scripts that touch many systems require multiple secrets and
+types of secrets particularly when orchestrating across different clouds.
+Best practice is to not hard code any secrets in scripts, but currently
+this requires custom code on different platforms to handle this securely.
+
+This RFC proposes new cmdlets to make this simpler and secure.
+
+## Motivation
+
+> As a PowerShell script author,
+> I can securely use multiple secrets,
+> so that I can automate complex orchestration across multiple remote resources.
+
+## High Level Design
+
+Secrets are stored securely in a local vault.
```

Comment:
To clarify, is the local vault going to use the Windows Credential vault or will this be standalone? I'm going to guess standalone to work everywhere.

If the Windows Credential vault is being considered as part of implementation on Windows, would need to consider ramifications of being able to expose secrets from that vault via powershell.

Reply:
It's already possible to access windows credential vault credentials with PowerShell with some inline C#.

Reply:
I guess that's fine if we're already within the user's context to access the credential vault in the first place. If the vault is in an 'unlocked' state, anything ran within the context is going to be able to get to it, same goes for other keyvault formats.

Perhaps the credential manager itself needs to be upgraded to require a password to unlock if it hasn't been used in a while or being polled for decrypting secrets instead of being directly passed to whatever may need to ingest it. However, that's outside the scope of this RFC.

Reply:
Yeah, and there are lots of modules for it (like my BetterCredentials module).

Reply:
Will add mention in Considerations section as out of scope for now